src/components/form/FormField.test.tsx

```diff
@@ -0,0 +1,126 @@
+import { SubmitHandler, UseFormRegister, useForm } from 'react-hook-form';
+
+import { render, screen, userEvent, waitFor } from '@/utils/renderers';
+
+import { FormField } from './FormField';
+
+interface FormValues {
+  name: string;
+  startDate: string;
+}
+
+// Mock form setup
+const MockForm = ({ onSubmit }: { onSubmit: SubmitHandler<FormValues> }) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <FormField<FormValues>
+        error={errors.name}
+        label="Name"
+        name="name" // Explicitly type this as 'name' from FormValues
+        placeholder="Enter your name"
+        register={register}
+        type="text"
+      />
+
+      <FormField<FormValues>
+        label="Start Date"
+        min="2024-09-01"
+        name="startDate" // Explicitly type this as 'startDate' from FormValues
+        register={register}
+        type="date"
+      />
+
+      <button type="submit">Submit</button>
```

Comment:
Do you prefer using a `submit` button as opposed to a listener on the button? As doesn't this end up causing a refresh?

Reply:
Not when using the react-hook-library and the `onSubmit` function on the form. When forms where first developed we used `action`, but since JavaScript has come into the picture there are workarounds to this, but we are now, with server-side components and actions, we are moving back to the old `action` method. That is good as JavaScript has created too many escape hatch clauses in how the HTML spec works.